core/scripts/gateway/client/send_request.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/ethereum/go-ethereum/crypto"
+
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/api"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/handlers/functions"
+	"github.com/smartcontractkit/chainlink/v2/core/services/s4"
+)
+
+func main() {
+	gatewayURL := flag.String("gateway_url", "", "Gateway URL")
+	privateKey := flag.String("private_key", "", "Private key to sign the message with")
+	messageId := flag.String("message_id", "", "Request ID")
+	methodName := flag.String("method", "", "Method name")
+	donId := flag.String("don_id", "", "DON ID")
+	s4SetSlotId := flag.Uint("s4_set_slot_id", 0, "S4 set slot ID")
+	s4SetVersion := flag.Uint64("s4_set_version", 0, "S4 set version")
+	s4SetExpiration := flag.Int64("s4_set_expiration", 0, "S4 set expiration")
+	s4SetPayload := flag.String("s4_set_payload", "", "S4 set payload")
+	flag.Parse()
+
+	// validate key and extract address
+	key, err := crypto.HexToECDSA(*privateKey)
+	if err != nil {
+		fmt.Println("error parsing private key", err)
+		return
+	}
+	address := crypto.PubkeyToAddress(key.PublicKey)
+
+	// build payload (if relevant)
+	var payloadJSON []byte
+	if *methodName == "secrets_set" {
+		envelope := s4.Envelope{
+			Address:    address.Bytes(),
+			SlotID:     *s4SetSlotId,
+			Version:    *s4SetVersion,
+			Payload:    []byte(*s4SetPayload),
+			Expiration: *s4SetExpiration,
+		}
+		signature, err := envelope.Sign(key)
+		if err != nil {
+			fmt.Println("error signing S4 envelope", err)
+			return
+		}
+
+		s4SetPayload := functions.SecretsSetRequest{
```

Comment:
I think this is valid separation of concern. Here it is a request, while Envelope serves different purpose.